Smoke test migration to behave

tisutisu opened the pull request:
This PR will cover story: https://issues.redhat.com/browse/HELM-378

changes also includes below improvements discussed:
```
1. __post_init__ should be called once before anything else and never called again.
2. Methods should be provided to update secrets as needed rather than calling __post_init__
3. No step defs should update workflow_test variables directly, workflow_test methods should be provided.
4. Step def files should not be setting chart name and report as part of workflow test creation.
5. Add logging for workflow including content of OWNERS file
```

mmulholla reviewed (changes requested):
Nice work. Have a few minor comments/questions.

tisutisu commented on tests/functional/behave_features/common/utils/chart_certification.py:
```diff
@@ -56,7 +56,9 @@ def set_git_username_email(self, repo, username, email):
         bot_name = os.environ.get("BOT_NAME")
+        logging.debug(f"Enviroment variable value BOT_NAME: {bot_name}")
         bot_token = os.environ.get("BOT_TOKEN")
+        logging.debug(f"Enviroment variable value BOT_TOKEN: {bot_token}")
```
Got catch. Removed the logging here for bot_token.

tisutisu commented on tests/functional/behave_features/common/utils/chart_certification.py:
```diff
@@ -367,18 +384,44 @@ def cleanup (self):
+        self.test_report = test_report
+        chart_name, chart_version = self.get_chart_name_version()
+        logging.debug(f"Got chart_name: {chart_name} and chart_version: {chart_version} from the report")
+        self.secrets.test_chart = self.test_chart
```
Good catch. Removed this update now.

tisutisu commented on tests/functional/behave_features/common/utils/chart_certification.py:
```diff
@@ -367,18 +384,44 @@ def cleanup (self):
+        chart_name, chart_version = self.get_chart_name_version()
+        logging.debug(f"Got chart_name: {chart_name} and chart_version: {chart_version} from the chart")
+        self.secrets.test_chart = self.test_chart
+        self.secrets.test_report = self.test_report
```
Good catch. Removed this update now. It is not needed.

tisutisu commented on tests/functional/behave_features/HC-06_provider_delivery_control.feature:
```diff
@@ -0,0 +1,48 @@
+      | vendor_type  | vendor    | report_path               | index_file                        | provider_control_owners | provider_control_report |
+      | partners     | hashicorp | tests/data/report.yaml    | unpublished-certified-charts.yaml | true                    | true                    |
+
+  @external-feedback @partners
```
@external-feedback is the tag used for identifying test cases which we have added as part of customer feedback. Earlier QE had a initiative to calculate percentage of tests which are being as part of customer feedback in the test suite. It is part of that. I think since the tag is applicable to all the examples of the scenario, so tagged at the scenario level to avoid tagging at each example. @mmulholla WDYT?

mmulholla replied:
Would rather see the @partner tag on the examples to be consistent with other tests. 
@external-feedback is fine but need to review tests to see which should have the tag.

tisutisu replied:
moved the `partners` tag to the example level to be consistent with other tests. @mmulholla since you are more closely working with chart-verifier, would require your help to identify what other tests we can mark as `@external-feedback`

tisutisu commented:
> Nice work. Have a few minor comments/questions.

Addressed the comments in latest commit 
3aaaf90 . Also responded the questions. @mmulholla Please have a look.

mmulholla reviewed (approved):
Approved, thx.
